### PR TITLE
Block multiplayer operations until beatmap selection completes

### PR DESCRIPTION
**Pull request by smoogipoo**
It turns out that you can can exit match song select while the loading overlay is visible, which returns you to the match subscreen while the request to the server is still ongoing. Currently this keeps the ready button enabled, because this isn't a blocking operation.

This change turns it into a blocking operation, disabling the ready button in the process.

Can be simulated via 
```diff
diff --git a/osu.Game/Online/Multiplayer/OnlineMultiplayerClient.cs b/osu.Game/Online/Multiplayer/OnlineMultiplayerClient.cs
index a3423d4189..083ea1385c 100644
--- a/osu.Game/Online/Multiplayer/OnlineMultiplayerClient.cs
+++ b/osu.Game/Online/Multiplayer/OnlineMultiplayerClient.cs
@@ -202,14 +202,15 @@ public override Task AddPlaylistItem(MultiplayerPlaylistItem item)
             return connection.InvokeAsync(nameof(IMultiplayerServer.AddPlaylistItem), item);
         }

-        public override Task EditPlaylistItem(MultiplayerPlaylistItem item)
+        public override async Task EditPlaylistItem(MultiplayerPlaylistItem item)
         {
             if (!IsConnected.Value)
-                return Task.CompletedTask;
+                return;

             Debug.Assert(connection != null);

-            return connection.InvokeAsync(nameof(IMultiplayerServer.EditPlaylistItem), item);
+            await Task.Delay(5000);
+            await connection.InvokeAsync(nameof(IMultiplayerServer.EditPlaylistItem), item);
         }

         public override Task RemovePlaylistItem(long playlistItemId)
```

### CODE REVIEW

**Review by bdach — CHANGES_REQUESTED** [review]
see above

**Comment by smoogipoo** [comment]
Should now be fixed :)

**Comment by bdach** [comment]
I can still break this by doing the following:

1. With the patch in the OP, edit an item
2. Back out to song select and click on the edit button again while the edit from step 1 hasn't finished
3. When the loading layer is still showing from the previous operation being in progress, click the osu! button in the lower right
4. The song select screen is now stuck after the loading layer disappears (can't drag, can't select via keyboard)

https://user-images.githubusercontent.com/20418176/174330703-7c58cd36-33f0-427c-bf5e-6262b84845ba.mp4

File: osu.Game/Online/Multiplayer/OnlineMultiplayerClient.cs [hunk 1]
```diff
@@ -202,14 +202,16 @@ public override Task AddPlaylistItem(MultiplayerPlaylistItem item)
...
+        public override async Task EditPlaylistItem(MultiplayerPlaylistItem item)
         {
             if (!IsConnected.Value)
-                return Task.CompletedTask;
+                return;
 
             Debug.Assert(connection != null);
 
-            return connection.InvokeAsync(nameof(IMultiplayerServer.EditPlaylistItem), item);
+            await Task.Delay(10000);
```
Comment (bdach) [inline osu.Game/Online/Multiplayer/OnlineMultiplayerClient.cs:212]: Presumably wasn't meant to be committed.

File: osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSongSelect.cs [hunk 2]
```diff
@@ -1,9 +1,13 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+#nullable enable
```
Comment (bdach) [inline osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSongSelect.cs:4]: Should be removed now after https://github.com/ppy/osu/pull/18743 (and is also the correct merge resolution).

**Review by bdach — APPROVED** [review]
Seems good now

**Pull request merged**